Update docs for tasks (#83)
Elektrobit/ebcl_template: docs/tests/performance.md

# Performance tests

The EBcL test library provides a very simple setup for performance tests.

> The idea of this performance tests are to implement a continous monitoring, and detecting changes early.

Reviewer: ```suggestion
The idea of those performance tests are to implement a continuous monitoring, and detecting changes early.
```

> It is not the idea to provide precise and stable performance measuements for the system under test.

Reviewer: ```suggestion
It is not the idea to provide precise and stable performance measurements for the system under test.
```

> The performance tests require speical images,

Reviewer: ```suggestion
The performance tests require special images,
```

These test overlays provide a quite simple way to extend images with test extensions
in a very controlled manner, and make it easy to fully understand the impact of the
test extensions. They may be also helpful for other kind of tests which require modified images.

In general, we expect that also test extensions and manipulations are done during image build,
and not during test execution time. If a test requires image manitpulations, a test specific
build target is created, which shall implement all changes in an easy to understand and
reproducible way.
This helps not only for the quality argumentation, but also helps in maintaining these modifications
when the images change over time, reduces the requirements for the test exection environment,
and allows to inspect the result of the manipulations.

The [performance tests implemented in qemu_performance.robot](https://github.com/Elektrobit/ebcl_template/robot_tests/qemu_performance.robot)
make use of the _Image_ class to build the test-specific images.
For a CI or hardware setup, this step needs to be separated,
but for the local QEMU build it helps to simplify the tests.

> For exectuing the tests, the _Performance_ class is used.

Reviewer: ```suggestion
For executing the tests, the _Performance_ class is used.
```
